sources/packages/web/src/components/common/NoticeOfAssessmentFormView.vue

```diff
@@ -1,36 +1,54 @@
 <template>
-  <formio formName="noticeofassessment" :data="initialData"></formio>
+  <formio-container formName="noticeOfAssessment" :formData="initialData" />
 </template>
 
 <script lang="ts">
-import { ref, watch } from "vue";
+import { watch, defineComponent, ref, onMounted } from "vue";
 import { StudentAssessmentsService } from "@/services/StudentAssessmentsService";
 import { AssessmentNOAAPIOutDTO } from "@/services/http/dto";
 
-export default {
+interface NoticeOfAssessment extends AssessmentNOAAPIOutDTO {
+  viewOnly?: boolean;
+}
+
+export default defineComponent({
+  emits: ["assessmentData"],
   props: {
     assessmentId: {
       type: Number,
       required: true,
     },
+    viewOnly: {
+      type: Boolean,
+      required: false,
+    },
   },
-  setup(props: any) {
-    const initialData = ref({} as AssessmentNOAAPIOutDTO);
+  setup(props, { emit }) {
+    const initialData = ref<NoticeOfAssessment>({} as NoticeOfAssessment);
```

Comment:
Just as recommendation, doing `ref<NoticeOfAssessment>` and `{} as NoticeOfAssessment` are not needed. The type will be defined with one or another either way.